src/server/ModelFlatFile.ts

```diff
@@ -0,0 +1,49 @@
+import type { Object } from 'ts-toolbelt';
+
+import type {
+  FetchFields,
+  FetchOpts,
+} from 'boardgame.io/dist/types/src/server/db/base';
+import { FlatFile } from 'boardgame.io/server';
+
+interface ModelFetchOpts extends FetchOpts {
+  model?: boolean;
+}
+
+interface ModelFetchFields extends FetchFields {
+  model: Record<string, unknown>; // TODO: improve
+}
+
+type ModelFetchResult<O extends ModelFetchOpts> = Object.Pick<
+  ModelFetchFields,
+  Object.SelectKeys<O, true>
+>;
+
+/**
+ * FlatFile data storage with support for saving a model.
+ */
+export class ModelFlatFile extends FlatFile {
+  async fetch<O extends ModelFetchOpts>(
+    matchID: string,
+    opts: O,
+  ): Promise<ModelFetchResult<O>> {
+    const result = await super.fetch(matchID, opts);
+    if (opts.model) {
+      const key = this.getModelKey(matchID);
+      return {
+        ...result,
+        // @ts-expect-error getItem is private, ask boardgame.io to make it protected
+        model: await this.getItem(key),
+      } as unknown as ModelFetchResult<O>;
+    }
+    return result as unknown as ModelFetchResult<O>;
+  }
+  async setModel(id: string, model: Record<string, unknown>): Promise<void> {
```

Comment:
Add a few blank lines in this class definition to make it better readable?  (Why doesn't the linter complain about it?)

Reply:
probably it's just not an active linting rule. If you want that, we can configure it.